### ISSUE TITLE
[mongo-cxx-driver] Build error

### ISSUE THREAD

**Issue by cPalum** [microsoft/vcpkg#32078]
Package: mongo-cxx-driver:x64-linux -> 3.7.0

**Host Environment**

- Host: x64-linux
- Compiler: GNU 13.1.1
-    vcpkg-tool version: 2023-06-15-8c254a5fb6c503615834fc66bd0717664a339a2e
    vcpkg-scripts version: 80ecf3249 2023-06-16 (15 hours ago)

**To Reproduce**

`vcpkg install mongo-cxx-driver`

**Failure logs**

```
-- Using cached mongodb-mongo-cxx-driver-r3.7.0.tar.gz.
-- Cleaning sources at /usr/bin/vcpkg/buildtrees/mongo-cxx-driver/src/r3.7.0-3713f74bb9.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source /usr/bin/vcpkg/downloads/mongodb-mongo-cxx-driver-r3.7.0.tar.gz
-- Applying patch fix-dependencies.patch
-- Applying patch fix-msvc-cxx17.patch
-- Using source at /usr/bin/vcpkg/buildtrees/mongo-cxx-driver/src/r3.7.0-3713f74bb9.clean
-- Configuring x64-linux
-- Building x64-linux-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:134 (message):
    Command failed: /usr/bin/cmake --build . --config Debug --target install -- -v -j5
    Working Directory: /usr/bin/vcpkg/buildtrees/mongo-cxx-driver/x64-linux-dbg
    See logs for more information:
      /usr/bin/vcpkg/buildtrees/mongo-cxx-driver/install-x64-linux-dbg-out.log

Call Stack (most recent call first):
  installed/x64-linux/share/vcpkg-cmake/vcpkg_cmake_build.cmake:74 (vcpkg_execute_build_process)
  installed/x64-linux/share/vcpkg-cmake/vcpkg_cmake_install.cmake:16 (vcpkg_cmake_build)
  ports/mongo-cxx-driver/portfile.cmake:37 (vcpkg_cmake_install)
  scripts/ports.cmake:147 (include)



```
<details><summary>/usr/bin/vcpkg/buildtrees/mongo-cxx-driver/install-x64-linux-dbg-out.log</summary>

```
[1/108] /usr/bin/c++ -DBSONCXX_STATIC -DBSON_STATIC -DMONGO_CXX_DRIVER_COMPILING -I/usr/bin/vcpkg/buildtrees/mongo-cxx-driver/src/r3.7.0-3713f74bb9.clean/src/bsoncxx/.. -I/usr/bin/vcpkg/buildtrees/mongo-cxx-driver/x64-linux-dbg/src/bsoncxx/.. -isystem /usr/bin/vcpkg/installed/x64-linux/include -fPIC -g -std=c++17 -fPIC -MD -MT src/bsoncxx/CMakeFiles/bsoncxx_static.dir/exception/error_code.cpp.o -MF src/bsoncxx/CMakeFiles/bsoncxx_static.dir/exception/error_code.cpp.o.d -o src/bsoncxx/CMakeFiles/bsoncxx_static.dir/exception/error_code.cpp.o -c /usr/bin/vcpkg/buildtrees/mongo-cxx-driver/src/r3.7.0-3713f74bb9.clean/src/bsoncxx/exception/error_code.cpp
[2/108] /usr/bin/c++ -DBSONCXX_STATIC -DBSON_STATIC -DMONGO_CXX_DRIVER_COMPILING -I/usr/bin/vcpkg/buildtrees/mongo-cxx-driver/src/r3.7.0-3713f74bb9.clean/src/bsoncxx/.. -I/usr/bin/vcpkg/buildtrees/mongo-cxx-driver/x64-linux-dbg/src/bsoncxx/.. -isystem /usr/bin/vcpkg/installed/x64-linux/include -fPIC -g -std=c++17 -fPIC -MD -MT src/bsoncxx/CMakeFiles/bsoncxx_static.dir/types.cpp.o -MF src/bsoncxx/CMakeFiles/bsoncxx_static.dir/types.cpp.o.d -o src/bsoncxx/CMakeFiles/bsoncxx_static.dir/types.cpp.o -c /usr/bin/vcpkg/buildtrees/mongo-cxx-driver/src/r3.7.0-3713f74bb9.clean/src/bsoncxx/types.cpp
[3/108] /usr/bin/c++ -DBSONCXX_STATIC -DBSON_STATIC -DMONGO_CXX_DRIVER_COMPILING -I/usr/bin/vcpkg/buildtrees/mongo-cxx-driver/src/r3.7.0-3713f74bb9.clean/src/bsoncxx/.. -I/usr/bin/vcpkg/buildtrees/mongo-cxx-driver/x64-linux-dbg/src/bsoncxx/.. -isystem /usr/bin/vcpkg/installed/x64-linux/include -fPIC -g -std=c++17 -fPIC -MD -MT src/bsoncxx/CMakeFiles/bsoncxx_static.dir/array/view.cpp.o -MF src/bsoncxx/CMakeFiles/bsoncxx_static.dir/array/view.cpp.o.d -o src/bsoncxx/CMakeFiles/bsoncxx_static.dir/array/view.cpp.o -c /usr/bin/vcpkg/buildtrees/mongo-cxx-driver/src/r3.7.0-3713f74bb9.clean/src/bsoncxx/array/view.cpp
[4/108] /usr/bin/c++ -DBSONCXX_STATIC -DBSON_STATIC -DMONGO_CXX_DRIVER_COMPILING -I/usr/bin/vcpkg/buildtrees/mongo-cxx-driver/src/r3.7.0-3713f74bb9.clean/src/bsoncxx/.. -I/usr/bin/vcpkg/buildtrees/mongo-cxx-driver/x64-linux-dbg/src/bsoncxx/.. -isystem /usr/bin/vcpkg/installed/x64-linux/include -fPIC -g -std=c++17 -fPIC -MD -MT src/bsoncxx/CMakeFiles/bsoncxx_static.dir/oid.cpp.o -MF src/bsoncxx/CMakeFiles/bsoncxx_static.dir/oid.cpp.o.d -o src/bsoncxx/CMakeFiles/bsoncxx_static.dir/oid.cpp.o -c /usr/bin/vcpkg/buildtrees/mongo-cxx-driver/src/r3.7.0-3713f74bb9.clean/src/bsoncxx/oid.cpp
[5/108] /usr/bin/c++ -DBSONCXX_STATIC -DBSON_STATIC -DMONGO_CXX_DRIVER_COMPILING -I/usr/bin/vcpkg/buildtrees/mongo-cxx-driver/src/r3.7.0-3713f74bb9.clean/src/bsoncxx/.. -I/usr/bin/vcpkg/buildtrees/mongo-cxx-driver/x64-linux-dbg/src/bsoncxx/.. -isystem /usr/bin/vcpkg/installed/x64-linux/include -fPIC -g -std=c++17 -fPIC -MD -MT src/bsoncxx/CMakeFiles/bsoncxx_static.dir/json.cpp.o -MF src/bsoncxx/CMakeFiles/bsoncxx_static.dir/json.cpp.o.d -o src/bsoncxx/CMakeFiles/bsoncxx_static.dir/json.cpp.o -c /usr/bin/vcpkg/buildtrees/mongo-cxx-driver/src/r3.7.0-3713f74bb9.clean/src/bsoncxx/json.cpp
[6/108] /usr/bin/c++ -DBSONCXX_STATIC -DBSON_STATIC -DMONGO_CXX_DRIVER_COMPILING -I/usr/bin/vcpkg/buildtrees/mongo-cxx-driver/src/r3.7.0-3713f74bb9.clean/src/bsoncxx/.. -I/usr/bin/vcpkg/buildtrees/mongo-cxx-driver/x64-linux-dbg/src/bsoncxx/.. -isystem /usr/bin/vcpkg/installed/x64-linux/include -fPIC -g -std=c++17 -fPIC -MD -MT src/bsoncxx/CMakeFiles/bsoncxx_static.dir/string/view_or_value.cpp.o -MF src/bsoncxx/CMakeFiles/bsoncxx_static.dir/string/view_or_value.cpp.o.d -o src/bsoncxx/CMakeFiles/bsoncxx_static.dir/string/view_or_value.cpp.o -c /usr/bin/vcpkg/buildtrees/mongo-cxx-driver/src/r3.7.0-3713f74bb9.clean/src/bsoncxx/string/view_or_value.cpp
[7/108] /usr/bin/c++ -DBSONCXX_STATIC -DBSON_STATIC -DMONGO_CXX_DRIVER_COMPILING -I/usr/bin/vcpkg/buildtrees/mongo-cxx-driver/src/r3.7.0-3713f74bb9.clean/src/bsoncxx/.. -I/usr/bin/vcpkg/buildtrees/mongo-cxx-driver/x64-linux-dbg/src/bsoncxx/.. -isystem /usr/bin/vcpkg/installed/x64-linux/include -fPIC -g -std=c++17 -fPIC -MD -MT src/bsoncxx/CMakeFiles/bsoncxx_static.dir/validate.cpp.o -MF src/bsoncxx/CMakeFiles/bsoncxx_static.dir/validate.cpp.o.d -o src/bsoncxx/CMakeFiles/bsoncxx_static.dir/validate.cpp.o -c /usr/bin/vcpkg/buildtrees/mongo-cxx-driver/src/r3.7.0-3713f74bb9.clean/src/bsoncxx/validate.cpp
[8/108] /usr/bin/c++ -DBSONCXX_STATIC -DBSON_STATIC -DMONGO_CXX_DRIVER_COMPILING -I/usr/bin/vcpkg/buildtrees/mongo-cxx-driver/src/r3.7.0-3713f74bb9.clean/src/bsoncxx/.. -I/usr/bin/vcpkg/buildtrees/mongo-cxx-driver/x64-linux-dbg/src/bsoncxx/.. -isystem /usr/bin/vcpkg/installed/x64-linux/include -fPIC -g -std=c++17 -fPIC -MD -MT src/bsoncxx/CMakeFiles/bsoncxx_static.dir/types/bson_value/value.cpp.o -MF src/bsoncxx/CMakeFiles/bsoncxx_static.dir/types/bson_value/value.cpp.o.d -o src/bsoncxx/CMakeFiles/bsoncxx_static.dir/types/bson_value/value.cpp.o -c /usr/bin/vcpkg/buildtrees/mongo-cxx-driver/src/r3.7.0-3713f74bb9.clean/src/bsoncxx/types/bson_value/value.cpp
[9/108] /usr/bin/c++ -DBSONCXX_STATIC -DBSON_STATIC -DMONGOCXX_STATIC -DMONGOC_STATIC -DMONGO_CXX_DRIVER_COMPILING -I/include -I/usr/bin/vcpkg/buildtrees/mongo-cxx-driver/src/r3.7.0-3713f74bb9.clean/src/mongocxx/.. -I/usr/bin/vcpkg/buildtrees/mongo-cxx-driver/x64-linux-dbg/src/mongocxx/.. -I/usr/bin/vcpkg/buildtrees/mongo-cxx-driver/src/r3.7.0-3713f74bb9.clean/src/bsoncxx/.. -I/usr/bin/vcpkg/buildtrees/mongo-cxx-driver/x64-linux-dbg/src/bsoncxx/.. -isystem /usr/bin/vcpkg/installed/x64-linux/include -fPIC -g -std=c++17 -fPIC -MD -MT src/mongocxx/CMakeFiles/mongocxx_static.dir/events/heartbeat_started_event.cpp.o -MF src/mongocxx/CMakeFiles/mongocxx_static.dir/events/heartbeat_started_event.cpp.o.d -o src/mongocxx/CMakeFiles/mongocxx_static.dir/events/heartbeat_started_event.cpp.o -c /usr/bin/vcpkg/buildtrees/mongo-cxx-driver/src/r3.7.0-3713f74bb9.clean/src/mongocxx/events/heartbeat_started_event.cpp
FAILED: src/mongocxx/CMakeFiles/mongocxx_static.dir/events/heartbeat_started_event.cpp.o 
/usr/bin/c++ -DBSONCXX_STATIC -DBSON_STATIC -DMONGOCXX_STATIC -DMONGOC_STATIC -DMONGO_CXX_DRIVER_COMPILING -I/include -I/usr/bin/vcpkg/buildtrees/mongo-cxx-driver/src/r3.7.0-3713f74bb9.clean/src/mongocxx/.. -I/usr/bin/vcpkg/buildtrees/mongo-cxx-driver/x64-linux-dbg/src/mongocxx/.. -I/usr/bin/vcpkg/buildtrees/mongo-cxx-driver/src/r3.7.0-3713f74bb9.clean/src/bsoncxx/.. -I/usr/bin/vcpkg/buildtrees/mongo-cxx-driver/x64-linux-dbg/src/bsoncxx/.. -isystem /usr/bin/vcpkg/installed/x64-linux/include -fPIC -g -std=c++17 -fPIC -MD -MT src/mongocxx/CMakeFiles/mongocxx_static.dir/events/heartbeat_started_event.cpp.o -MF src/mongocxx/CMakeFiles/mongocxx_static.dir/events/heartbeat_started_event.cpp.o.d -o src/mongocxx/CMakeFiles/mongocxx_static.dir/events/heartbeat_started_event.cpp.o -c /usr/bin/vcpkg/buildtrees/mongo-cxx-driver/src/r3.7.0-3713f74bb9.clean/src/mongocxx/events/heartbeat_started_event.cpp
In file included from /usr/bin/vcpkg/buildtrees/mongo-cxx-driver/src/r3.7.0-3713f74bb9.clean/src/mongocxx/events/heartbeat_started_event.cpp:15:
/usr/bin/vcpkg/buildtrees/mongo-cxx-driver/src/r3.7.0-3713f74bb9.clean/src/mongocxx/../mongocxx/events/heartbeat_started_event.hpp:54:10: error: ‘uint16_t’ in namespace ‘std’ does not name a type; did you mean ‘wint_t’?
   54 |     std::uint16_t port() const;
      |          ^~~~~~~~
      |          wint_t
/usr/bin/vcpkg/buildtrees/mongo-cxx-driver/src/r3.7.0-3713f74bb9.clean/src/mongocxx/events/heartbeat_started_event.cpp:33:15: error: no declaration matches ‘uint16_t mongocxx::v_noabi::events::heartbeat_started_event::port() const’
   33 | std::uint16_t heartbeat_started_event::port() const {
      |               ^~~~~~~~~~~~~~~~~~~~~~~
/usr/bin/vcpkg/buildtrees/mongo-cxx-driver/src/r3.7.0-3713f74bb9.clean/src/mongocxx/events/heartbeat_started_event.cpp:33:15: note: no functions named ‘uint16_t mongocxx::v_noabi::events::heartbeat_started_event::port() const’
/usr/bin/vcpkg/buildtrees/mongo-cxx-driver/src/r3.7.0-3713f74bb9.clean/src/mongocxx/../mongocxx/events/heartbeat_started_event.hpp:33:20: note: ‘class mongocxx::v_noabi::events::heartbeat_started_event’ defined here
   33 | class MONGOCXX_API heartbeat_started_event {
      |                    ^~~~~~~~~~~~~~~~~~~~~~~
[10/108] /usr/bin/c++ -DBSONCXX_STATIC -DBSON_STATIC -DMONGOCXX_STATIC -DMONGOC_STATIC -DMONGO_CXX_DRIVER_COMPILING -I/include -I/usr/bin/vcpkg/buildtrees/mongo-cxx-driver/src/r3.7.0-3713f74bb9.clean/src/mongocxx/.. -I/usr/bin/vcpkg/buildtrees/mongo-cxx-driver/x64-linux-dbg/src/mongocxx/.. -I/usr/bin/vcpkg/buildtrees/mongo-cxx-driver/src/r3.7.0-3713f74bb9.clean/src/bsoncxx/.. -I/usr/bin/vcpkg/buildtrees/mongo-cxx-driver/x64-linux-dbg/src/bsoncxx/.. -isystem /usr/bin/vcpkg/installed/x64-linux/include -fPIC -g -std=c++17 -fPIC -MD -MT src/mongocxx/CMakeFiles/mongocxx_static.dir/events/heartbeat_failed_event.cpp.o -MF src/mongocxx/CMakeFiles/mongocxx_static.dir/events/heartbeat_failed_event.cpp.o.d -o src/mongocxx/CMakeFiles/mongocxx_static.dir/events/heartbeat_failed_event.cpp.o -c /usr/bin/vcpkg/buildtrees/mongo-cxx-driver/src/r3.7.0-3713f74bb9.clean/src/mongocxx/events/heartbeat_failed_event.cpp
FAILED: src/mongocxx/CMakeFiles/mongocxx_static.dir/events/heartbeat_failed_event.cpp.o 
/usr/bin/c++ -DBSONCXX_STATIC -DBSON_STATIC -DMONGOCXX_STATIC -DMONGOC_STATIC -DMONGO_CXX_DRIVER_COMPILING -I/include -I/usr/bin/vcpkg/buildtrees/mongo-cxx-driver/src/r3.7.0-3713f74bb9.clean/src/mongocxx/.. -I/usr/bin/vcpkg/buildtrees/mongo-cxx-driver/x64-linux-dbg/src/mongocxx/.. -I/usr/bin/vcpkg/buildtrees/mongo-cxx-driver/src/r3.7.0-3713f74bb9.clean/src/bsoncxx/.. -I/usr/bin/vcpkg/buildtrees/mongo-cxx-driver/x64-linux-dbg/src/bsoncxx/.. -isystem /usr/bin/vcpkg/installed/x64-linux/include -fPIC -g -std=c++17 -fPIC -MD -MT src/mongocxx/CMakeFiles/mongocxx_static.dir/events/heartbeat_failed_event.cpp.o -MF src/mongocxx/CMakeFiles/mongocxx_static.dir/events/heartbeat_failed_event.cpp.o.d -o src/mongocxx/CMakeFiles/mongocxx_static.dir/events/heartbeat_failed_event.cpp.o -c /usr/bin/vcpkg/buildtrees/mongo-cxx-driver/src/r3.7.0-3713f74bb9.clean/src/mongocxx/events/heartbeat_failed_event.cpp
In file included from /usr/bin/vcpkg/buildtrees/mongo-cxx-driver/src/r3.7.0-3713f74bb9.clean/src/mongocxx/events/heartbeat_failed_event.cpp:15:
/usr/bin/vcpkg/buildtrees/mongo-cxx-driver/src/r3.7.0-3713f74bb9.clean/src/mongocxx/../mongocxx/events/heartbeat_failed_event.hpp:54:10: error: ‘int64_t’ in namespace ‘std’ does not name a type
   54 |     std::int64_t duration() const;
      |          ^~~~~~~
/usr/bin/vcpkg/buildtrees/mongo-cxx-driver/src/r3.7.0-3713f74bb9.clean/src/mongocxx/../mongocxx/events/heartbeat_failed_event.hpp:68:10: error: ‘uint16_t’ in namespace ‘std’ does not name a type; did you mean ‘wint_t’?
   68 |     std::uint16_t port() const;
      |          ^~~~~~~~
      |          wint_t
/usr/bin/vcpkg/buildtrees/mongo-cxx-driver/src/r3.7.0-3713f74bb9.clean/src/mongocxx/events/heartbeat_failed_event.cpp:35:14: error: no declaration matches ‘int64_t mongocxx::v_noabi::events::heartbeat_failed_event::duration() const’
   35 | std::int64_t heartbeat_failed_event::duration() const {
      |              ^~~~~~~~~~~~~~~~~~~~~~
/usr/bin/vcpkg/buildtrees/mongo-cxx-driver/src/r3.7.0-3713f74bb9.clean/src/mongocxx/events/heartbeat_failed_event.cpp:35:14: note: no functions named ‘int64_t mongocxx::v_noabi::events::heartbeat_failed_event::duration() const’
/usr/bin/vcpkg/buildtrees/mongo-cxx-driver/src/r3.7.0-3713f74bb9.clean/src/mongocxx/../mongocxx/events/heartbeat_failed_event.hpp:33:20: note: ‘class mongocxx::v_noabi::events::heartbeat_failed_event’ defined here
   33 | class MONGOCXX_API heartbeat_failed_event {
      |                    ^~~~~~~~~~~~~~~~~~~~~~
/usr/bin/vcpkg/buildtrees/mongo-cxx-driver/src/r3.7.0-3713f74bb9.clean/src/mongocxx/events/heartbeat_failed_event.cpp:45:15: error: no declaration matches ‘uint16_t mongocxx::v_noabi::events::heartbeat_failed_event::port() const’
   45 | std::uint16_t heartbeat_failed_event::port() const {
      |               ^~~~~~~~~~~~~~~~~~~~~~
/usr/bin/vcpkg/buildtrees/mongo-cxx-driver/src/r3.7.0-3713f74bb9.clean/src/mongocxx/events/heartbeat_failed_event.cpp:45:15: note: no functions named ‘uint16_t mongocxx::v_noabi::events::heartbeat_failed_event::port() const’
/usr/bin/vcpkg/buildtrees/mongo-cxx-driver/src/r3.7.0-3713f74bb9.clean/src/mongocxx/../mongocxx/events/heartbeat_failed_event.hpp:33:20: note: ‘class mongocxx::v_noabi::events::heartbeat_failed_event’ defined here
   33 | class MONGOCXX_API heartbeat_failed_event {
      |                    ^~~~~~~~~~~~~~~~~~~~~~
[11/108] /usr/bin/c++ -DBSONCXX_STATIC -DBSON_STATIC -DMONGOCXX_STATIC -DMONGOC_STATIC -DMONGO_CXX_DRIVER_COMPILING -I/include -I/usr/bin/vcpkg/buildtrees/mongo-cxx-driver/src/r3.7.0-3713f74bb9.clean/src/mongocxx/.. -I/usr/bin/vcpkg/buildtrees/mongo-cxx-driver/x64-linux-dbg/src/mongocxx/.. -I/usr/bin/vcpkg/buildtrees/mongo-cxx-driver/src/r3.7.0-3713f74bb9.clean/src/bsoncxx/.. -I/usr/bin/vcpkg/buildtrees/mongo-cxx-driver/x64-linux-dbg/src/bsoncxx/.. -isystem /usr/bin/vcpkg/installed/x64-linux/include -fPIC -g -std=c++17 -fPIC -MD -MT src/mongocxx/CMakeFiles/mongocxx_static.dir/events/command_started_event.cpp.o -MF src/mongocxx/CMakeFiles/mongocxx_static.dir/events/command_started_event.cpp.o.d -o src/mongocxx/CMakeFiles/mongocxx_static.dir/events/command_started_event.cpp.o -c /usr/bin/vcpkg/buildtrees/mongo-cxx-driver/src/r3.7.0-3713f74bb9.clean/src/mongocxx/events/command_started_event.cpp
[12/108] /usr/bin/c++ -DBSONCXX_STATIC -DBSON_STATIC -DMONGOCXX_STATIC -DMONGOC_STATIC -DMONGO_CXX_DRIVER_COMPILING -I/include -I/usr/bin/vcpkg/buildtrees/mongo-cxx-driver/src/r3.7.0-3713f74bb9.clean/src/mongocxx/.. -I/usr/bin/vcpkg/buildtrees/mongo-cxx-driver/x64-linux-dbg/src/mongocxx/.. -I/usr/bin/vcpkg/buildtrees/mongo-cxx-driver/src/r3.7.0-3713f74bb9.clean/src/bsoncxx/.. -I/usr/bin/vcpkg/buildtrees/mongo-cxx-driver/x64-linux-dbg/src/bsoncxx/.. -isystem /usr/bin/vcpkg/installed/x64-linux/include -fPIC -g -std=c++17 -fPIC -MD -MT src/mongocxx/CMakeFiles/mongocxx_static.dir/events/heartbeat_succeeded_event.cpp.o -MF src/mongocxx/CMakeFiles/mongocxx_static.dir/events/heartbeat_succeeded_event.cpp.o.d -o src/mongocxx/CMakeFiles/mongocxx_static.dir/events/heartbeat_succeeded_event.cpp.o -c /usr/bin/vcpkg/buildtrees/mongo-cxx-driver/src/r3.7.0-3713f74bb9.clean/src/mongocxx/events/heartbeat_succeeded_event.cpp
[13/108] /usr/bin/c++ -DBSONCXX_STATIC -DBSON_STATIC -DMONGOCXX_STATIC -DMONGOC_STATIC -DMONGO_CXX_DRIVER_COMPILING -I/include -I/usr/bin/vcpkg/buildtrees/mongo-cxx-driver/src/r3.7.0-3713f74bb9.clean/src/mongocxx/.. -I/usr/bin/vcpkg/buildtrees/mongo-cxx-driver/x64-linux-dbg/src/mongocxx/.. -I/usr/bin/vcpkg/buildtrees/mongo-cxx-driver/src/r3.7.0-3713f74bb9.clean/src/bsoncxx/.. -I/usr/bin/vcpkg/buildtrees/mongo-cxx-driver/x64-linux-dbg/src/bsoncxx/.. -isystem /usr/bin/vcpkg/installed/x64-linux/include -fPIC -g -std=c++17 -fPIC -MD -MT src/mongocxx/CMakeFiles/mongocxx_static.dir/gridfs/bucket.cpp.o -MF src/mongocxx/CMakeFiles/mongocxx_static.dir/gridfs/bucket.cpp.o.d -o src/mongocxx/CMakeFiles/mongocxx_static.dir/gridfs/bucket.cpp.o -c /usr/bin/vcpkg/buildtrees/mongo-cxx-driver/src/r3.7.0-3713f74bb9.clean/src/mongocxx/gridfs/bucket.cpp
ninja: build stopped: subcommand failed.
```
</details>

**Additional context**

**Comment by JonLiu1993** [comment]
Thanks for posting this issue. I saw you use cache `-- mongodb-mongo-cxx-driver-r3.7.0.tar.gz` to install `mongo-cxx-driver`, could you please remove cache and rebuild `mongo-cxx-driver`. I can install it succeed locally.
````
-- Installing: /home/test/vcpkg/packages/mongo-cxx-driver_x64-linux/share/libmongocxx/libmongocxx-config.cmake
-- Installing: /home/test/vcpkg/packages/mongo-cxx-driver_x64-linux/share/mongo-cxx-driver/usage
-- Installing: /home/test/vcpkg/packages/mongo-cxx-driver_x64-linux/share/mongo-cxx-driver/copyright
-- Performing post-build validation
Elapsed time to handle mongo-cxx-driver:x64-linux: 58 s
Total install time: 1.5 min
mongo-cxx-driver provides CMake targets:

    find_package(bsoncxx CONFIG REQUIRED)
    target_link_libraries(main PRIVATE $<IF:$<TARGET_EXISTS:mongo::bsoncxx_static>,mongo::bsoncxx_static,mongo::bsoncxx_shared>)

    find_package(mongocxx CONFIG REQUIRED)
    target_link_libraries(main PRIVATE $<IF:$<TARGET_EXISTS:mongo::mongocxx_static>,mongo::mongocxx_static,mongo::mongocxx_shared>)

````
````

**Comment by dg0yt** [comment]
> I saw you use cache `-- mongodb-mongo-cxx-driver-r3.7.0.tar.gz` to install `mongo-cxx-driver`

This is the source asset. No need to remove it.

I guess it is the frequenly seen issue with gcc 13 being stricter about includes.

**Comment by JonLiu1993** [comment]
@cPalum, My local linux gcc version is `11.3.0` and this error does not reproduce. This error is related to the standards of different versions of gcc, and may have nothing to do with vcpkg. so I temporarily close this issue, if you have other questions I can reopen it.

**Comment by Korisss** [comment]
I also have this problem with clang 12 (maybe it's because of some changes in the headers). But I noticed that this has been [fixed in version 3.7.2](https://github.com/mongodb/mongo-cxx-driver/pull/961) and it would be nice if someone update mongo-cxx-driver
